tasking/pulp/tasking/scheduler.py

```diff
@@ -260,110 +251,34 @@ def tick(self):
         old_timestamp = datetime.utcnow() - timedelta(seconds=constants.CELERYBEAT_LOCK_MAX_AGE)
 
         # Updating the current lock if lock is on this instance of celerybeat
-        result = CeleryBeatLock.objects(celerybeat_name=self.celerybeat_name).\
-            update(set__timestamp=datetime.utcnow())
-
-        # If current instance has lock and updated lock_timestamp, call super
-        if result == 1:
+        try:
+            celerybeat_lock = TaskLock.objects.get(name=self.celerybeat_name)
+            celerybeat_lock.timestamp = timezone.now()
+            celerybeat_lock.save()
+            # If current instance has lock and updated lock_timestamp, call super
             _logger.debug(_('Lock updated by %(celerybeat_name)s')
                           % {'celerybeat_name': self.celerybeat_name})
             ret = self.call_tick(self, self.celerybeat_name)
-        else:
-            # check for old enough time_stamp and remove if such lock is present
-            CeleryBeatLock.objects(timestamp__lte=old_timestamp).delete()
+        except TaskLock.DoesNotExist:
+            TaskLock.objects(name=self.celerybeat_name, timestamp__lte=old_timestamp).delete()
             try:
-                lock_timestamp = datetime.utcnow()
-
+                lock_timestamp = timezone.now()
                 # Insert new lock entry
-                new_lock = CeleryBeatLock(celerybeat_name=self.celerybeat_name,
-                                          timestamp=lock_timestamp)
-                new_lock.save()
+                TaskLock.objects.create(name=self.celerybeat_name, timestamp=lock_timestamp,
```

Comment:
The `timestamp` will be set to now() automatically so it's not needed and it can be removed from L#265 too.